.github/workflows/manual-build.yml

```diff
@@ -0,0 +1,84 @@
+# Deploy the app to stable. Not builds for development/QA
+name: Manually build latest stable 
+on: 
+  workflow_dispatch: # manually run the action
+
+env:
+  XCODE_VERSION: "14.0"
+  GITHUB_CONTEXT: ${{ toJSON(github) }} # used by fastlane for deploying app 
+  FASTLANE_XCODEBUILD_SETTINGS_RETRIES: 6 # increase timeout for xcodebuild settings. We have seen this command take a bit longer to do on CI server. 
+
+jobs:
+  get-latest-git-tag:
+    name: Get latest git tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get latest git tag
+        uses: octokit/request-action@v2.1.4
+        id: get_latest_release
+        with:
+          route: GET /repos/customerio/amiapp-flutter/releases/latest
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      # Checkout the git tag right before building since older git tags might not have the necessary files
+      # that the newer CI config asks for. Do all CI environment prep before checking out the git tag.
+      - run: echo "LATEST_GIT_TAG=${{ fromJson(steps.get_latest_release.outputs.data).tag_name }}" >> $GITHUB_ENV
+      - name: Checkout latest git tag
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.LATEST_GIT_TAG }}
+          clean: false # to not delete the files created in environment setup step
```

Comment:
I am now realizing that this will not work here. 

GH Actions jobs run as sandboxed environments. Meaning that the git checkout here will not impact other jobs such as `deploy-android-app`. 

Instead, I suggest that this job has an [output](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idoutputs) and then each future job checks out the tag. 

```yml
jobs:
  get-latest-git-tag:
    name: Get latest git tag
    runs-on: ubuntu-latest"
    outputs:
      latest-tag-name: ${{ fromJson(steps.get_latest_release.outputs.data).tag_name }}
    steps:
      - name: Get latest git tag
        uses: octokit/request-action@v2.1.4
        id: get_latest_release
        with:
          route: GET /repos/customerio/amiapp-flutter/releases/latest
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

deploy-android-app:
    runs-on: ubuntu-latest
    needs: get-latest-git-tag
    steps:
      - uses: actions/checkout@v3
        with:
          ref: ${{ needs.get-latest-git-tag.outputs.latest-tag-name }}
```

Reply:
That's a fair point, even though they would be all on the same commit/tag it doesn't ensure its the latest one without output.